### PR TITLE
Solve accuracy problem for inplace operators when MKL-DNN mode is enabled

### PR DESCRIPTION
**Pull request by kbinias**
This PR is important to improve SE-ResNeXt accuracy when MKL-DNN mode is enabled.

test=develop

### CODE REVIEW

File: paddle/fluid/framework/operator.cc [hunk 1]
```diff
@@ -760,7 +760,8 @@ void OperatorWithKernel::RunImpl(const Scope& scope,
 
   kernel_iter->second(ExecutionContext(*this, exec_scope, *dev_ctx));
 
-  if (!transfered_inplace_vars.empty()) {
```
Comment (mozga-intel) [inline paddle/fluid/framework/operator.cc:763]: The comment is required.

**Review by mozga-intel — DISMISSED** [review]

File: paddle/fluid/framework/operator.cc (hunk 1 above)
Comment (kbinias) [reply paddle/fluid/framework/operator.cc:763]: Done

**Comment by luotao1** [comment]
image
I create a pre-run CE job to test this PR. It may have 6 hours (2+hours to wait, 3+hours to run) to get the result.
http://ce.paddlepaddle.org:8080/viewLog.html?buildId=5726&buildTypeId=build_PaddlePrDebugBuild2&tab=buildLog

**Comment by luotao1** [comment]
```
[06:54:57][Step 1/1] Task [resnet50_net_GPU] failed!
[06:54:57][Step 1/1] details: [cifar10_128_AllReduce_GPU_4_Cards_train_speed] failed, diff ratio: 0.14169066041564812 larger than 0.08.
[06:54:57][Step 1/1] Task [resnet50_net_GPU] failed!
[06:54:57][Step 1/1] details: [cifar10_128_Reduce_GPU_4_Cards_train_speed] failed, diff ratio: 0.10477659900593321 larger than 0.08.
```
http://ce.paddlepaddle.org:8080/viewLog.html?buildId=5726&tab=buildLog&buildTypeId=build_PaddlePrDebugBuild2&logTab=tree&filter=all
```
[06:15:00][Step 1/1] Task [resnet50] failed!
[06:15:00][Step 1/1] details: [cifar10_128_train_speed] failed, diff ratio: 0.06417423636124564 larger than 0.06.
[06:15:00][Step 1/1] evaluation kpi info: False ['[cifar10_128_train_acc] pass', '[cifar10_128_train_speed] failed, diff ratio: 0.06417423636124564 larger than 0.06.', '[cifar10_128_gpu_memory] pass
```

**Comment by kbinias** [comment]
@luotao1 Thanks for your tests. We can suspend this PR. The purpose of this PR is only to resolve the problem for SE-ResNeXt accuracy (decrease from 77 to 43). We're working on improving accuracy drop for ResNet-50.

**Comment by luotao1** [comment]
@kbinias 
- All the tests in CE don't run on MKLDNN, they are run on CPU and GPU now.
- You can rerun this test by yourself after you make a fix.

**Comment by kbinias** [comment]
We will merge this PR in our next PR which will solve the problem with accuracy for ResNet-50

**Pull request closed without merging**